### PR TITLE
Add support of mTLS in Spring Cloud Config Client

### LINKED ISSUE #21512: Add support of mTLS in Spring Cloud Config Client
### Describe the bug

I'd like to use Spring Cloud Config Client to connect to our Spring Cloud Config Server using mTLS. But after providing corresponding truststore.ks an keystore.ks in spring-cloud-config section I figured out that keystore.ks is used as a parameter for webClientOptions.setTrustStoreOptions instead of webClientOptions.setKeyStoreOptions. So no client certs could be sent to server. From my point of view current behavior looks like a bug.

### Expected behavior

keystore.ks used as a parameter for webClientOptions.setKeyStoreOptions

### Actual behavior

keystore.ks used as a parameter for webClientOptions.setTrustStoreOptions

### How to Reproduce?

Setup Spring Cloud Config Client with truststore and keystore and try to connect Spring Cloud Config Server with mTLS enabled. No client certs will be provided.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.4.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ishumov**
Fixes #21512

### CODE REVIEW

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java [hunk 1]
```diff
@@ -77,15 +77,16 @@ public static WebClient createHttpClient(Vertx vertx, SpringCloudConfigClientCon
...
+                Path keyStorePath = springCloudConfig.keyStore.get();
+                String type = determineStoreType(keyStorePath);
+                KeyStoreOptionsBase storeOptions = storeOptions(keyStorePath, springCloudConfig.keyStorePassword,
                         createStoreOptions(type));
                 if (isPfx(type)) {
-                    webClientOptions.setPfxTrustOptions((PfxOptions) storeOptions);
+                    webClientOptions.setPfxKeyCertOptions((PfxOptions) storeOptions);
                 } else {
-                    webClientOptions.setTrustStoreOptions((JksOptions) storeOptions);
+                    webClientOptions.setKeyStoreOptions((JksOptions) storeOptions);
```
Comment (gsmet) [inline extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java:89]: Could you explain the change of method? Does it still work in the initial case covered?

**Review by gsmet — COMMENTED** [review]
Thanks for this. I added a small question inline.

File: extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java (hunk 1 above)
Comment (ishumov) [reply extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java:89]: Do you mean SSL session without checking client cert on a server? Usual way in Java is using truststore for this purpose. In that case keystore not needed. From my point of view using keystore as a store for CA certificate is a mistake.

For this change I was guided by https://vertx.io/docs/vertx-core/java/#ssl

Comment (gsmet) [reply extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java:89]: @michalszynkiewicz you probably have more experience than me on the Vert.x web client, could you have a look?

Comment (michalszynkiewicz) [reply extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java:89]: This looks good to me.
Setting trust store options from keystore didn't look good.

**Review by michalszynkiewicz — APPROVED** [review]

**Review by geoand — APPROVED** [review]

**Pull request merged**